distributed/tests/test_events.py

```diff
@@ -0,0 +1,165 @@
+import pickle
+
+from distributed import Event
+from distributed.utils_test import gen_cluster
+from distributed.utils_test import client, cluster_fixture, loop  # noqa F401
+
+
+@gen_cluster(client=True, nthreads=[("127.0.0.1", 8)] * 2)
+async def test_event(c, s, a, b):
+    def wait_for_it_failing(x):
+        event = Event("x")
+
+        # Event is not set in another task so far
+        assert not event.wait(timeout=0.05)
+        assert not event.is_set()
+
+    def wait_for_it_ok(x):
+        event = Event("x")
+
+        # Event is set in another task
+        assert event.wait(timeout=0.5)
+        assert event.is_set()
+
+    def set_it():
+        event = Event("x")
+        event.set()
+
+    def clear_it():
+        event = Event("x")
+        event.clear()
+
+    wait_futures = c.map(wait_for_it_failing, range(10))
+    await c.gather(wait_futures)
+
+    set_future = c.submit(set_it)
+    await c.gather(set_future)
+
+    wait_futures = c.map(wait_for_it_ok, range(10))
+    await c.gather(wait_futures)
+
+    clear_future = c.submit(clear_it)
+    await c.gather(clear_future)
+
+    wait_futures = c.map(wait_for_it_ok, range(10))
+    await c.gather(wait_futures)
+
+    assert not s.extensions["events"]._events
+    assert not s.extensions["events"]._waiter_count
+
+
+def test_default_event(client):
+    event = Event("x")
+    assert not event.is_set()
```

Comment:
With small tests like these we prefer using the async API if you're comfortable with it.  They are easier to debug if something goes wrong and they're also much faster to run, which helps to keep the overall test suite duration down.

I usually test everything in async mode, and then have one or two tests specifically to make sure that we're calling `client.sync` correctly.

Reply:
Thanks for the suggestion! I did not understood this distinction in other tests in the first place, but it makes a lot of sense.